examples/single_agent.py

```diff
@@ -35,7 +35,9 @@ def act(self, obs: Observation):
         )
 
 
-def main(scenarios, headless, num_episodes, seed, max_episode_steps=None):
+def main(
+    scenarios, headless, num_episodes, seed, authkey=b"secret", max_episode_steps=None
+):
```

Comment:
Please change the default for `authkey` to be `None`. A value like `"secret"` is a bit arbitrary.

Also please rename to `auth_key` since our convention splits by "_" between words.

Reply:
+ I have renamed all `authkey` occurrences with `auth_key`.
+ `auth_key` should be of type byte string. `None` is of type `NoneType`. Hence, I have changed default `auth_key` to be `b"None"`.

Reply:
My opinion here is that the internal requirements should not show up in the API. I cannot see any particular reason why the `auth_key` argument should require a byte string at the interface level. Perhaps `None` could be replaced with an empty string but unless there is some compelling reason I do not think we should be using a byte string at this level.

Reply:
We should really use the `NoneType` `None` here. 

From the docs: https://docs.python.org/3/library/multiprocessing.html#multiprocessing.connection.Listener

> If authkey is given and not None, it should be a byte string and will be used as the secret key for an HMAC-based authentication challenge. No authentication is done if authkey is None. AuthenticationError is raised if authentication fails. 

So we do really want "no authentication" here on the client side, and that is enabled with `None` as the authkey

Reply:
Changed default `auth_key` to `None`.